File: mobile/lib/dtos/order_dto.dart
import 'package:json_annotation/json_annotation.dart';

import '../core/entities/order_status.dart';
import 'address_dto.dart';
import 'order_item_dto.dart';

part 'order_dto.g.dart';

@JsonSerializable()
class OrderDTO {
  String? id;
  int? orderNumber;
  List<OrderItemDTO>? items;
  OrderStatus? status;
  AddressDTO? shippingAddress;
  double? total;

  OrderDTO({
    this.id,
    this.orderNumber,
    this.items,
    this.status,
    this.shippingAddress,
    this.total,
  });

  factory OrderDTO.fromJson(Map<String, dynamic> json) => _$OrderDTOFromJson(json);

  Map<String, dynamic> toJson() => _$OrderDTOToJson(this);
}

File: bff/src/main/kotlin/me/lucasbatista/vienna/sdk/dto/OrderDTO.kt
package me.lucasbatista.vienna.sdk.dto

data class OrderDTO(
    var id: String? = null,
    var orderNumber: Int? = null,
    var items: List<OrderItemDTO>? = null,
    var status: OrderStatus? = null,
    var shippingAddress: AddressDTO? = null,
    var total: Double? = null,
)
